File: pset2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// پروتوتایپ توابع
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // گرفتن متن از کاربر
    string text = get_string("Text: ");

    // شمارش تعداد حروف، کلمات و جملات در متن
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // محاسبه شاخص Coleman-Liau
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // چاپ سطح کلاس
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", (int) round(index));
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1; // شروع از 1 برای شمارش اولین کلمه
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: pset6/Readability.py
from cs50 import get_string

def main():
    text = get_string("Text: ")

    # Initialize counters
    letters = 0
    words = 0
    sentences = 0

    # Analyze the text
    for char in text:
        if char.isalpha():
            letters += 1
        elif char.isspace():
            continue
        elif char in ".!?":
            sentences += 1

    # Words are counted by splitting the text by whitespace
    words = len(text.split())

    # Calculate the averages L and S
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # Calculate the Coleman-Liau index
    index = 0.0588 * L - 0.296 * S - 15.8

    # Determine the grade level
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")

if __name__ == "__main__":
    main()
